File: CodingTest/Python/solution16.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42627
def solution(jobs):
    answer = 0
    import heapq
    jobs.sort()
    print(jobs)
    cnt = len(jobs)
    time = 0
    temp = []
    while True:
        while jobs and time >= jobs[0][0]:
            at, rt = jobs.pop(0)
            heapq.heappush(temp, (rt, at))
        if temp:
            rt, at = heapq.heappop(temp)
            answer += (time - at + rt)
            time += rt
        else:
            time += 1

        if not jobs and not temp:
            break
    return answer // cnt

File: CodingTest/Java/Solution16.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12985
package CodingTest.Java;

public class Solution16 {
    public int solution(int n, int a, int b) {
        int answer = 1;
        a -= 1;
        b -= 1;

        while (a != b) {
            a = a / 2;
            b = b / 2;
            
            answer += 1;
        }
        return answer - 1;
    }
}